pyiceberg/io/pyarrow.py

```diff
@@ -1170,7 +1167,7 @@ def project_table(
     if len(tables) < 1:
         return pa.Table.from_batches([], schema=schema_to_pyarrow(projected_schema, include_field_ids=False))
 
-    result = pa.concat_tables(tables)
+    result = pa.concat_tables(tables, promote_options="permissive")
```

Comment:
Docs for `promote_options="permissive"`
https://arrow.apache.org/docs/python/generated/pyarrow.concat_tables.html